examples/example_glfw_vulkan/main.cpp

```diff
@@ -68,33 +69,77 @@ static VKAPI_ATTR VkBool32 VKAPI_CALL debug_report(VkDebugReportFlagsEXT flags,
 }
 #endif // IMGUI_VULKAN_DEBUG_REPORT
 
-static void SetupVulkan(const char** extensions, uint32_t extensions_count)
+// Check if the enumration extension has to be enabled
+bool checkExtensions(VkInstance instance)
+{
+    uint32_t extensionCount = 0;
+    vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount, nullptr);
+    std::vector<VkExtensionProperties> availableExtensions(extensionCount);
+    vkEnumerateInstanceExtensionProperties(nullptr, &extensionCount, availableExtensions.data());
+
+    bool enablePortabilitySubset = false;
+    bool portabilitySubsetEnabled = false;
+    bool enableGetPhysicalDeviceProperties2 = false;
+    bool getPhysicalDeviceProperties2Enabled = false;
+
+#if defined(__APPLE__)
+    const bool macOS = true;
+#else
+    const bool macOS = false;
+#endif
+
+    for (const auto &extension : availableExtensions)
+    {
+        if (strcmp(extension.extensionName, VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME) == 0)
+        {
+            portabilitySubsetEnabled = true;
+        }
+        else if (strcmp(extension.extensionName, VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME) == 0)
+        {
+            getPhysicalDeviceProperties2Enabled = true;
+        }
+    }
+
+    if (portabilitySubsetEnabled && macOS)
+    {
+        uint32_t vulkanVersion;
+        vkEnumerateInstanceVersion(&vulkanVersion);
+        if (VK_VERSION_MAJOR(vulkanVersion) >= 1 && VK_VERSION_MINOR(vulkanVersion) >= 3 && VK_VERSION_PATCH(vulkanVersion) >= 216)
+        {
+            portabilitySubsetEnabled = false;
```

Comment:
Why is this check disabling portability subset? If the extension is available, it must be enabled.

Reply:
I understand your point of view. Now that I'm looking over everything again, I see that it would probably be easier to create two functions. One that checks the flag for `VK_INSTANCE_CREATE_ENUMERATE_PORTABILITY_BIT_KHR` and one that checks if the `VK_KHR_PORTABILITY_ENUMERATION_EXTENSION_NAME` is enabled. And of course both check if the OS is macOS and the Vulkan version is >= 1.3.216. Is this the best solution? Maybe you know a better one. I am always grateful for better ideas.

Reply:
Stop making unnecessary checks. What does the OS or driver version has to do with anything here? Read what the extension specification actually says, and follow that to the letter.

Reply:
There were some unnecessary checks in the old code. I'll give u that. I updated the code with a new command. But we need to check the OS and the Vulkan version, because under 1.3.216 it is not necessary to enable the enumeration extension. And all that is only needed on macOS… Or did I not understand your concern? I'm sry if I didn't understand you correctly.